D-lama-service/d-lama-service/DataProcessing/JasonDataParser.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System.Text.Json;
+
+namespace d_lama_service.DataProcessing
+{
+    public class JsonDataParser : IDataParser
+    {
+        private readonly string[] permittedExtensions = { ".json" };
+
+        public bool IsValidFormat(IFormFile file)
```

Comment:
as the method looks exactly as the one from TextDataParser, maybe make DataSetReader abstract with already implementing this behavior. And then just override the permittedExtensions property.

Reply:
Have a look at the new version from my last commit. Thanks!